src/Sulu/Bundle/WebsiteBundle/Command/DumpSitemapCommand.php

```diff
@@ -144,82 +176,61 @@ private function dumpSitemap(PortalInformation $portalInformation)
      */
     private function dumpProviderSitemap($alias, PortalInformation $portalInformation)
     {
-        $provider = $this->sitemapProviderPool->getProvider($alias);
-        $maxPage = $provider->getMaxPage();
+        $page = 1;
+        do {
+            $sitemap = $this->sitemapRenderer->renderSitemap(
+                $alias,
+                $page,
+                $portalInformation->getLocale(),
+                $portalInformation->getPortal(),
+                $portalInformation->getHost(),
+                $this->scheme
+            );
 
-        $pathFormat = '/%s/sitemaps/%s-%s.xml';
-        for ($page = 1; $page <= $maxPage; ++$page) {
-            $path = sprintf($pathFormat, $portalInformation->getUrl(), $alias, $page);
-            $this->dumpFile($path, $this->renderSitemap($alias, $page, $portalInformation));
-        }
+            $this->dumpFile(
```

Comment:
And why don't we use the `$maxPage` variable you have deleted before? Then it could simply be `while($page < $pageCount)` or something similar.